plugins/techdocs-module-mkdocs/src/components/MkDocsContent/MkDocsContent.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import { useApi, configApiRef } from '@backstage/core-plugin-api';
+
+import {
+  useTechDocsReader,
+  TechDocsShadowDom,
+  TechDocsShadowDomHooks,
+} from '@backstage/plugin-techdocs';
+
+import {
+  StylesTransformer,
+  ScriptTransformer,
+  ScrollTransformer,
+  HeaderTransformer,
+  FooterTransformer,
+  DrawerTransformer,
+  SidebarTransformer,
+  AnchorTransformer,
+  ImageTransformer,
+  SourceTransformer,
+  CodeTransformer,
+} from '../MkDocsContentTransformers';
+
+import { beforeSanitizeElements, afterSanitizeAttributes } from './hooks';
+
+/**
+ * Renders TechDocs content using MkDocs and ShadowRoot
+ * @public
+ */
+export const TechDocsReaderContent = () => {
+  const configApi = useApi(configApiRef);
+  const { content, onReady } = useTechDocsReader();
+
+  if (!content) {
+    return null;
+  }
+
+  const config = {
+    ADD_TAGS: ['link'],
+    FORBID_TAGS: ['style'],
+  };
+
+  const hooks: TechDocsShadowDomHooks = {
+    afterSanitizeAttributes,
+  };
+
+  const sanitizer = configApi.getOptionalConfig('techdocs.sanitizer');
+  const allowedHosts = sanitizer?.getOptionalStringArray('allowedIframeHosts');
+
+  if (allowedHosts) {
+    config.ADD_TAGS.push('iframe');
+    hooks.beforeSanitizeElements = beforeSanitizeElements(allowedHosts);
+  }
```

Comment:
This seems to be something that needs to happen no matter what type of content you decide to render in Backstage, did you see patterns of more of these things when you developed this module that could live in some sort of base reader that can be reused across multiple content components? Just thinking if these things need to be duplicated every time we will end up with different solutions?